packages/build/tests/core/tests.js

```diff
@@ -323,6 +323,30 @@ test('Print warning when headers file is missing from publish directory', async
   })
 })
 
+test.serial('Successfully builds ES module function with feature flag', async (t) => {
+  const spy = sinon.spy(zipItAndShipIt, 'zipFunctions')
+
+  await runFixture(t, 'functions_es_modules', {
+    flags: { featureFlags: { buildbot_es_modules_esbuild: true } },
+  })
+
+  const { args: callArgs } = spy.getCall(0)
+  t.true(callArgs[2].featureFlags.defaultEsModulesToEsbuild)
+
+  zipItAndShipIt.zipFunctions.restore()
```

Comment:
[pebble] We might want to put this line (and the other `restore()` below) in a `finally` block so that, if the test throws, `zipFunctions()` is still restored for the other tests.